### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by StudioEtrange** [jacebrowning/gitman#132]
Due to some shells call, we cannot use gitman on windows platform

**Comment by StudioEtrange** [comment]
For now : I am working on it, on this fork
https://github.com/StudioEtrange/gitman

**Comment by jacebrowning** [comment]
Can you paste the error messages you are seeing?

**Comment by StudioEtrange** [comment]
It is fixed on this commit on this fork : 
https://github.com/StudioEtrange/gitman/commit/3b2ca4b6e1aa0f4adb17b48f0899fbc38b44d61b

Mainly errors come from mkdir, rmdir commands

**Comment by jacebrowning** [comment]
I'd still like `gitman` to give the appearance of calling shell commands, so this is the appropriate place to put your logic: https://github.com/jacebrowning/gitman/blob/52e4fb1b4e5dd4b98122c78eaa00abee4286d4de/gitman/shell.py#L31-L32

There should also be a warning when symlink creation is skipped (call `log.warning()`).

**Comment by StudioEtrange** [comment]
If I put logic on line L31 and L32
There is a trap: function call is called with specific unix argument

For example here '-p' is revelant only on unix. On windows, it crash

```
call('mkdir', '-p', path)
```

So, If you really wish to keep the call to call function AND to put the logic on line L31-32, I will have too move all specific os option inside

Then It should looks like something like this:


```
def call(name, *args, _show=True, _ignore=False):
    program = CMD_PREFIX + ' '.join([name, *args])
    if _show:
        common.show(program)
    else:
        log.debug(program)

    if name == 'mkdir':
       if os.name=nt':
              # prepare a windows shell command
      else
              # prepare a linux shell command

       command = subprocess.run(
           [name, *args], universal_newlines=True,
           stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
    )

call("mkdir",path)
```


But, I dont understand why not just use python os module directly 
with :
```
os.mkdir(path)
```

**Comment by jacebrowning** [comment]
I need to think about the structure some more.

If you can get a PR passing CI, I will consider merging it and refactor the code later. Thanks!

**Comment by StudioEtrange** [comment]
Ok.
I have done my PR.

which fails on Travis...

I have to tweak some test :)

It will be done !

**Comment by jacebrowning** [comment]
Added in https://github.com/jacebrowning/gitman/pull/134.

**Comment by jacebrowning** [comment]
Please test `GitMan==1.1b2` if you have time.

**Comment by StudioEtrange** [comment]
I am on it

**Comment by jacebrowning** [comment]
I went ahead and released `GitMan==1.1`.

If you notice any issues on Windows (other than the CI tests), please file a bug and/or PR.
